CoffeaAnalysis/HNLAnalysis/HNLAnalysis_tte.py

```diff
@@ -0,0 +1,190 @@
+import numpy as np
+import awkward as ak
+from coffea import processor
+
+from CoffeaAnalysis.HNLAnalysis.helpers import save_anatuple_common, save_anatuple_lepton, save_anatuple_tau, save_Event_bjets
+from CoffeaAnalysis.HNLAnalysis.correction_helpers import compute_sf_e, compute_sf_tau, get_trigger_correction_tau
+from CoffeaAnalysis.HNLAnalysis.helpers import IsoElectron_mask, FinalTau_sel, delta_r, bjet_candidates
+from CoffeaAnalysis.HNLAnalysis.HNLProcessor import HNLProcessor
+
+class HNLAnalysis_tte(processor.ProcessorABC, HNLProcessor):
+    def __init__(self, stitched_list, tag, xsecs):
+        HNLProcessor.__init__(self, stitched_list, tag, xsecs)
+        acc_dict = {}
+        self.selections = self.get_selections()
+        for selection in self.selections:
+            acc_dict[f'n_ev_{selection}'] = processor.defaultdict_accumulator(int)
+            acc_dict[f'sumw_{selection}'] = processor.defaultdict_accumulator(float)
+        self._accumulator = processor.dict_accumulator(acc_dict)
+
+        #the corresponding data sample for tte channel (HLT=DoubleTau)
+        self.dataHLT = 'Tau_2018'
+
+    @property
+    def accumulator(self):
+        return self._accumulator
+
+    @staticmethod
+    def get_selections():
+        return [
+            'init',
+            'reweight',
+            'MET_Filter',
+            'AtLeast2tau1e',
+            'NoAdditionalIsoElectron',
+            'noIsoMuon',
+            'HLT',
+            'eSelection',
+            'drTau_e',
+            'drTau2Leptons',
+            'corrections'
+        ]
+
+    # we will receive a NanoEvents
+    def process(self, events):
+
+        out = self.accumulator.identity()
+        events, out, ds, mode = self.init_process(out, events)
+
+        # cut specific for that channel 
+        self.cut_mu_iso = 0.15 # veto events with tight mu isolation for ortogonality in signal region for channels with muon
+        self.cut_tau_idVSe = 6 # require tight isolation against electron for channel with reco electron
+        # trigger requirement 
+        self.cut_tau_pt = 40.  
+        self.cut_tau_eta = 2.1 
```

Comment:
we can have a single electron trigger -> don't modify tau cuts, but request it during trigger matching (as in other cases)